File: 2015/day4/deno/logic_test.ts
import { assertEquals } from "https://deno.land/std@0.208.0/assert/mod.ts";
import { doesHashStartWithFiveZeros, getLowestNumber } from "./logic.ts";

Deno.test("doesHashStartWithFiveZeros", () => {
    assertEquals(true, doesHashStartWithFiveZeros("000001dbbfa"));
    assertEquals(false, doesHashStartWithFiveZeros("000011dbbfa"));
});

Deno.test("getLowestNumber", async () => {
    assertEquals(609043, await getLowestNumber("abcdef"));
    assertEquals(1048970, await getLowestNumber("pqrstuv"));
});

File: 2015/day4/ruby/logic_test.rb
require 'test/unit'
require './logic'

class MoverTest < Test::Unit::TestCase
  def test_does_hash_start_with_five_zeros
    assert_equal(true, does_hash_start_with_five_zeros("000001"))
    assert_equal(false, does_hash_start_with_five_zeros("000011"))
  end

  def test_get_lowest_number
    assert_equal(609043, get_lowest_number("abcdef"))
    assert_equal(1048970, get_lowest_number("pqrstuv"))
  end
end
